cpp/dynamic_community_detection_module/algorithm/dynamic_community_detection.cpp

```diff
@@ -0,0 +1,370 @@
+#include <cmath>
+#include <mg_graph.hpp>
+
+#include "dynamic_community_detection.hpp"
+
+namespace LabelRankT {
+/* #region helper_methods */
+///@brief Checks if vector is subset of another vector.
+///
+///@tparam T -- set element data type
+///
+///@param v1 -- subset candidate
+///@param v2 -- vector against which the check is performed
+///
+///@return -- whether v1 is subset of v2
+template <class T>
+bool IsSubset(std::vector<T> const& v1, std::vector<T> const& v2) {
+  for (const auto& element : v1) {
+    if (std::find(v2.begin(), v2.end(), element) == v2.end()) return false;
+  }
+
+  return true;
+}
+
+std::unordered_set<std::uint64_t> LabelRankT::InNeighborsMemgraphIDs(
+    std::uint64_t node_id) {
+  std::unordered_set<std::uint64_t> neighbors;
+
+  auto neighbor_set = this->is_directed
+                          ? graph->InNeighbours(graph->GetInnerNodeId(node_id))
+                          : graph->Neighbours(graph->GetInnerNodeId(node_id));
+  for (const auto node_i : neighbor_set) {
```

Comment:
I’ve verified that the size is indeed 16 bytes. Since this is now resolved, could you DM me with the reasoning for this decision? I’m curious :slightly_smiling_face:

Reply:
I can share the details here: the reasoning is on modern x64 CPUs the parameters are passed in the registers if they are not bigger than the size of 2 registers. [Godbolted](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:2,lang:c%2B%2B,selection:(endColumn:137,endLineNumber:34,positionColumn:137,positionLineNumber:34,selectionStartColumn:137,selectionStartLineNumber:34,startColumn:137,startLineNumber:34),source:'%23include+%3Ccstdint%3E%0A%0Astruct+A+%7B%0A++++int64_t+a%3B%0A++++int64_t+b%3B%0A%7D%3B%0A%0Astruct+B+%7B%0A++++int64_t+a%3B%0A++++int64_t+b%3B%0A++++int64_t+c%3B%0A%7D%3B%0A%0Avoid+foo(int64_t+i)%3B%0A%0Avoid+foo(A+a)%3B%0A%0Avoid+foo(B+b)%3B%0A%0Avoid+foo2(B%26+b)%3B%0A%0A//+int64_t+is+passed+in+registers+for+sure%0Avoid+barI()+%7B%0A++++int64_t+i%3B%0A++++foo(i)%3B%0A%7D%0A%0A//+Without+knowing+the+binary+code,+you+can+see+that+A+is+copied+in+the+same+way+as+a+value+of+int64_t+%0Avoid+barA()+%7B%0A++++A+a%3B%0A++++foo(a)%3B%0A%7D%0A%0A//+But+when+a+function+called+with+B+passed+by+value,+you+can+see+the+function+call+itself+requires+more+preparation+in+the+machine+code%0Avoid+barB()+%7B%0A++++B+b%3B%0A++++foo(b)%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%232',t:'0')),k:49.66057441253264,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:clang_trunk,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'0',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:2,lang:c%2B%2B,libs:!(),options:'-std%3Dc%2B%2B17+-O1+-DNDEBUG',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:2,tree:'1'),l:'5',n:'0',o:'x86-64+clang+(trunk)+(C%2B%2B,+Editor+%232,+Compiler+%232)',t:'0')),header:(),l:'4',m:75.30864197530865,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compiler:2,editor:2,fontScale:14,fontUsePx:'0',wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+clang+(trunk)+(Compiler+%232)',t:'0')),k:48.205332011952024,l:'4',m:24.691358024691358,n:'0',o:'',s:0,t:'0')),k:50.33942558746736,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4) (you can try to change the optimization flags from `O1`  to `O2` or `O3`.